core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -684,13 +670,20 @@ private IndexResult innerIndex(Index index) throws IOException {
             final IndexResult indexResult;
             if (checkVersionConflictResult.isPresent()) {
                 indexResult = checkVersionConflictResult.get();
+                // norelease: this is not correct as this does not force an fsync, and we need to handle failures including replication
+                if (indexResult.hasFailure() || seqNo == SequenceNumbersService.UNASSIGNED_SEQ_NO) {
```

Comment:
I pushed 8b0e5014b1e29976f9dd5acf11422b3d7873ef2f to revert the change in preparation for merging master in after #22626 lands there.